crates/amaru-ledger/src/rules/transaction/vkey_witness.rs

```diff
@@ -0,0 +1,196 @@
+                        e, output,
+                    ))
+                })?;
+
+                if let Some(key_hash) = address.key_hash() {
+                    pkhs.push(key_hash);
+                };
+            };
+        }
+
+        pkhs
+    };
+
+    let withdrawal_pkhs = transaction_body
+        .withdrawals
+        .as_ref()
+        .map(|withdrawals| {
+            withdrawals
+                .iter()
+                .filter_map::<Hash<28>, _>(|(reward_account, _)| {
+                    // The first four bits of the reward account are 1110 for a key hash and 1111 for a script hash
+                    if reward_account[0] & 0b00010000 == 0 {
+                        Some(Hash::from(&reward_account[1..29]))
+                    } else {
+                        None
+                    }
+                })
+                .collect::<Vec<_>>()
+        })
+        .unwrap_or_default();
+
+    let vote_pkhs = transaction_body
+        .voting_procedures
+        .as_ref()
+        .map(|voting_procedures| {
+            voting_procedures
+                .iter()
+                .filter_map(|(voter, _)| match voter {
+                    Voter::ConstitutionalCommitteeKey(hash) => Some(*hash),
+                    Voter::DRepKey(hash) => Some(*hash),
+                    Voter::StakePoolKey(hash) => Some(*hash),
+                    Voter::ConstitutionalCommitteeScript(_) => None,
+                    Voter::DRepScript(_) => None,
+                })
+                .collect::<Vec<_>>()
+        })
+        .unwrap_or_default();
+
+    let certificate_pkhs = transaction_body
+        .certificates
+        .as_deref()
+        .map(|certificates| {
+            certificates
+                .iter()
+                .filter_map(|certificate| match certificate {
+                    Certificate::StakeRegistration(_) => None,
+                    Certificate::StakeDeregistration(stake_credential) => {
+                        stake_credential.key_hash()
+                    }
+                    Certificate::StakeDelegation(stake_credential, _) => {
+                        stake_credential.key_hash()
+                    }
+                    Certificate::PoolRegistration {
+                        operator,
+                        vrf_keyhash: _,
+                        pledge: _,
+                        cost: _,
+                        margin: _,
+                        reward_account: _,
+                        pool_owners: _,
+                        relays: _,
+                        pool_metadata: _,
+                    } => Some(*operator),
+                    Certificate::PoolRetirement(hash, _) => Some(*hash),
+                    Certificate::Reg(stake_credential, coin) => {
+                        if coin == &0 {
+                            None
+                        } else {
+                            stake_credential.key_hash()
+                        }
```

Comment:
That looks suspicious?

Reply:
It does, it was based off of the comment found [here](https://github.com/IntersectMBO/cardano-ledger/blob/81637a1c2250225fef47399dd56f80d87384df32/eras/conway/impl/src/Cardano/Ledger/Conway/TxCert.hs#L698). We don't have an `Option` for coin here though, so not sure how that case would be handled

Reply:
It's worth porting that comment :)